src/main/java/org/ohdsi/webapi/shiro/PermissionManager.java

```diff
@@ -176,18 +236,49 @@ public UserEntity registerUser(final String login, final String name, final User
 
     RoleEntity personalRole = this.addRole(login, false);
     this.addUser(user, personalRole, userOrigin, null);
+    addRolesForUser(login, userOrigin, user, newUserRoles, false);
+    addDefaultRolesForUser(login, userOrigin, user, defaultRoles);
+    // get user again, fresh from db with all new roles:
+    user = userRepository.findOne(user.getId());
+    return user;
+  }
 
-    if (defaultRoles != null) {
-      for (String roleName: defaultRoles) {
-        RoleEntity defaultRole = this.getSystemRoleByName(roleName);
-        if (defaultRole != null) {
-          this.addUser(user, defaultRole, userOrigin, null);
+  private void addRolesForUser(String login, UserOrigin userOrigin, UserEntity user, Set<String> roles, boolean isSystemRole) {
+    if (roles != null) {
+      for (String roleName: roles) {
+        // Temporary patch/workaround (in reality the role should have been added by sysadmin?):
+        if (!isSystemRole) {
+          pocAddUserRole(roleName);
+        }
+        // end temporary patch
```

Comment:
remove?

Reply:
as of now, this will only be called as part of `updateUser`, as part of `TeamProjectBasedAuthorizingFilter`, which only calls it after Arborist check on the (single) team project role... so it is not a concern in practice. But we need to decide whether we want roles being created on the fly in any way here...maybe not.